pkg/restore/restore.go

```diff
@@ -664,6 +853,22 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 				continue
 			}
 
+			// See if the PV already exists and we need to wait for it delete
+			fromCluster, err := ctx.pvClient.Get(name, metav1.GetOptions{})
+			if err != nil && apierrors.IsNotFound(err) {
+				addArkError(&errs, fmt.Errorf("error getting copy of persistent volume %q from cluster: %v", name, err))
+			}
+
+			if isPVReleasedOrBound(fromCluster) {
+				ctx.log.Infof("Persistent volume %q still in use, waiting for deletion", name)
+				ctx.pvWaiter.addDeleteChan(name)
+				_, err := ctx.pvWaiter.waitForDelete(name, ctx.namespaceTimeout)
```

Comment:
probably need to do a similar double-check on whether the PV's been deleted here, since it could've been deleted between the first check and adding the channel